packages/@aws-cdk/aws-iam/lib/role.ts

```diff
@@ -439,3 +439,18 @@ function validateMaxSessionDuration(duration?: number) {
     throw new Error(`maxSessionDuration is set to ${duration}, but must be >= 3600sec (1hr) and <= 43200sec (12hrs)`);
   }
 }
+
+/**
+ * A PolicyStatement that doesn't normalize its Principal field.
+ */
+class NonPrincipalNormalizingStatement extends PolicyStatement {
+  public toStatementJson(): any {
+    const stat = super.toStatementJson();
+
+    if (stat.Principal && stat.Principal === '*') {
```

Comment:
Can this be simplified to just `stat.Principal === '*'`? If `stat.Principal` is `undefined`, that will return `false`, like it does now.